gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -57,6 +62,9 @@
     this.featureSheetState = new MutableLiveData<>();
     this.activeProject = LiveDataReactiveStreams.fromPublisher(dataRepository.getActiveProject());
     this.navigator = navigator;
+    this.featureProcessor = BehaviorProcessor.create();
+
+    disposeOnClear(featureProcessor.flatMap(feature -> this.dataRepository.addFeature(feature).toFlowable()).subscribe(this::onFeatureAdded));
```

Comment:
Hm, let's chat about this. I might be confused about the number of active subscriptions here. I thought in this case we only have the one subscription, since we never actually subscribe to the processor/subject, only to the resulting transformation after calling `flatMap`. `flatMap` may still not be the right choice here, I'm still developing an understanding of these operators, I mostly just made them type check this round.

Reply:
Scratch that, I see what you're referring to now!